conans/test/functional/toolchain/test_make.py

```diff
@@ -0,0 +1,306 @@
+import os
+import platform
+import textwrap
+import unittest
+
+from nose.plugins.attrib import attr
+
+from conans.test.utils.tools import TestClient
+
+from parameterized.parameterized import parameterized
+
+
+@attr("toolchain")
+class Base(unittest.TestCase):
+
+    conanfile = textwrap.dedent("""
+        from conans import ConanFile, MakeToolchain
+        class App(ConanFile):
+            settings = "os", "arch", "compiler", "build_type"
+            requires = "hello/0.1"
+            generators = "make"
+            options = {"shared": [True, False], "fPIC": [True, False]}
+            default_options = {"shared": False, "fPIC": True}
+            def toolchain(self):
+                tc = MakeToolchain(self)
+                tc.definitions["SOME_DEFINITION"] = "SomeValue"
+                return tc
+
+
+            def build(self):
+                self.run("make -C .. app")
+
+        """)
+
+    app_h = textwrap.dedent("""
+        #pragma once
+        #ifdef WIN32
+          #define APP_LIB_EXPORT __declspec(dllexport)
+        #else
+          #define APP_LIB_EXPORT
+        #endif
+        APP_LIB_EXPORT void app();
+        """)
+
+    app_lib_cpp = textwrap.dedent("""
+        #include <iostream>
+        #include "app.h"
+        #include "hello.h"
+        void app() {
+            std::cout << "Hello: " << HELLO_MSG <<std::endl;
+            #ifdef NDEBUG
+            std::cout << "App: Release!" <<std::endl;
+            #else
+            std::cout << "App: Debug!" <<std::endl;
+            #endif
+            std::cout << "SOME_DEFINITION: " << SOME_DEFINITION << "\\n";
+        }
+        """)
+
+    app = textwrap.dedent("""
+        #include "app.h"
+        int main() {
+            app();
+        }
+        """)
+
+    makefile = textwrap.dedent("""
+        include build/conan_toolchain.mak
+        include build/conanbuildinfo.mak
+        ifndef CONAN_TOOLCHAIN_INCLUDED
+            $(error >> Not using toolchain)
+        endif
+
+        $(info >> CONAN_BUILD_TYPE: $(CONAN_BUILD_TYPE))
+        $(info >> CONAN_TC_CFLAGS: $(CONAN_TC_CFLAGS))
+        $(info >> CONAN_TC_CXXFLAGS: $(CONAN_TC_CXXFLAGS))
+        $(info >> CONAN_TC_DEFINES: $(CONAN_TC_DEFINES))
+        $(info >> CONAN_TC_LDFLAGS: $(CONAN_TC_LDFLAGS))
+
+
+        CFLAGS += $(CONAN_TC_CFLAGS)
```

Comment:
I think this probably should go to the generated toolchain. The purpose is that using it is as transparent as possible, and require very little modifications to the consumer makefiles. Then, the toolchain could have options to change the behavior, like do not actually changing the CXXFLAGS (and let users do it themselves), but probably the default behavior could default to define them directly.

Reply:
this must have been left in accidentally. i will remove.

Reply:
There is now a function called `CONAN_TC_SETUP` which users can call as follows:
`$(call CONAN_TC_SETUP)`

This will do the tedious and error prone work of properly appending all the relevant `CONAN_` variables into the relevant "standard" variables. This is a big improvement, and I have copied the same strategy and submitted a PR for the `make` generator.